w3-aggregation.md

```diff
@@ -0,0 +1,337 @@
+
+```mermaid
+sequenceDiagram
+participant Storefront as 🌐<br/><br/>did:web:web3.storage
+    participant Authority as 🌐<br/><br/>did:web:spade.storage
+
+    Storefront->>Authority: invoke `aggregate/get`
+```
+
+## Capabilities
+
+This section describes the capabilities that form the w3 aggregation protocol, along with the details relevant for invoking capabilities with a service provider.
+
+In this document, we will be looking at `spade.storage` as an implementer of the `aggregate/*` and `offer/*` protocol.
+
+### `aggregate/offer`
+
+A Storefront principal can invoke a capabilty to submit an aggregate ready for offers.
+
+> `did:web:web3.storage` invokes capability from `did:web:spade.storage`
+
+```json
+{
+  "iss": "did:web:web3.storage",
+  "aud": "did:web:spade.storage",
+  "att": [{
+    "with": "did:web:web3.storage",
+    "can": "aggregate/offer",
+    "nb": {
+      "offer": {
+        "link": "bagy...aggregate",
+        "size": 110101,
+        "commP": "commP...",
+        "src": ["https://w3s.link/ipfs/bagy...aggregate"]
+      }
+    }
+  }],
+  "prf": [],
+  "sig": "..."
+}
+```
+
+This capability is invoked to submit a request to a broker service when an aggregate is ready for a Filecoin deal. The `nb.offer.link` is the CAR CID that refers to a "Ferry" aggregate, a collection of dag-cbor blocks with format:
+
+```json
+{
+  "link": "bag...",
+  "size": 110101,
+  "commP": "commP...",
+  "src": ["https://.../bag(...).car"]
+}
+```
+
+A receipt will be generated to acknowledge the received request. This receipt MUST contain an [effect](https://github.com/ucan-wg/invocation/#7-effect) with a subsequent task (`.fx.join` field) that is run when submitted aggregate is processed and either succeeds (implying that aggregate was accepted and deals will be arranged) or fail (with `error` describing a problem with an aggregate)
+
+```json
+{
+  "ran": "bafy...invocation",
+  "out": {
+    "ok": {
+      "status": "queued"
+    }
+  },
+  "fx": {
+    "join": { "/": "bafy...dequeue" }
+  },
+  "meta": {},
+  "iss": "did:web:spade.storage",
+  "prf": []
+}
+```
+
+Open questions:
+
+- can we get `commP` of the aggregate with `commP` of every CAR that is part of it?
+
+### `aggregate/get`
+
+A Storefront principal can invoke a capability to get state of an accepted aggregate.
+
+> `did:web:web3.storage` invokes capability from `did:web:spade.storage`
+
+```json
+{
+  "iss": "did:web:web3.storage",
+  "aud": "did:web:spade.storage",
+  "att": [{
+    "with": "did:web:web3.storage",
+    "can": "aggregate/get",
+    "nb": {
+      "commP": "commP...",
+    }
+  }],
+  "prf": [],
+  "sig": "..."
+}
+```
+
+### `offer/review`
+
+When a broker receives an `aggregate/offer` invocation from a Storefront Principal, an [Effect](https://github.com/ucan-wg/invocation/#7-effect) for this submission is created with join task to be performed asynchronously.
```

Comment:
Not sure I understand this part - is this the delegation that `bafy...dequeue` referrs to in `fx.join`?

Reply:
yes!

Reply:
Since in last call we decided offer reviews is not really a thing and instead spade just attempts to arrange them and either succeeds or fails I don't think previously suggested `offer/review` still makes sense, probably should be called `offer/arrange` instead ?